### PR TITLE
Log user page access and errors.

### PR DESCRIPTION
**Pull request by andorp**

### CODE REVIEW

**Comment by andorp** [comment]
Partly fixes: #915

**Comment by andorp** [comment]
We should introduce a new type class to transform values to log message parts and hide sensual information from. I would do that in a different pull request.

The snippet below shows a debug level log:
````
2016-08-01 15:46:06.777596 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:06.778015 UTC [WARN] [anonym] Not logged in
2016-08-01 15:46:06.782899 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:06.791325 UTC [INFO] [anonym] access to page PageHomeWithLoginPrompt
2016-08-01 15:46:09.885089 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:09.885829 UTC [DEBUG] [xd1yDlnwPX0bUAeRK172WePz] form processed for PageHomeWithLoginPrompt
2016-08-01 15:46:09.894083 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:09.895378 UTC [INFO] [xd1yDlnwPX0bUAeRK172WePz] access to page PageUserSettings
2016-08-01 15:46:09.895728 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:09.896001 UTC [INFO] [xd1yDlnwPX0bUAeRK172WePz] gets access to FormPageRep PageUserSettings
2016-08-01 15:46:11.590972 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:11.593275 UTC [INFO] [xd1yDlnwPX0bUAeRK172WePz] gets access to PageOverviewOfSpaces
2016-08-01 15:46:12.362875 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:12.38825 UTC [INFO] [xd1yDlnwPX0bUAeRK172WePz] gets access to PageOverviewOfWildIdeas
2016-08-01 15:46:13.747441 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:13.776825 UTC [INFO] [xd1yDlnwPX0bUAeRK172WePz] gets access to ViewIdea
2016-08-01 15:46:18.758726 UTC [DEBUG] daemon [logger] recieved a message.
2016-08-01 15:46:18.760478 UTC [ERROR] [xd1yDlnwPX0bUAeRK172WePz] PersistError404 {persistErrorMessage = "Idea"}
````

File: src/Action.hs [hunk 1]
```diff
@@ -382,13 +395,15 @@ validLoggedIn us = isJust (us ^. usUserId) && isJust (us ^. usSessionToken)
...
+getSpacesForCurrentUser :: (ActionUserHandler m, ActionPersist m, ActionLog m) => m [IdeaSpace]
 getSpacesForCurrentUser = do
     user <- currentUser
     query $ getSpacesForRoles (user ^. userRoleSet)
 
-deleteUser :: (ActionPersist m) => AUID User -> m ()
-deleteUser = update . DeactivateUser
+deleteUser :: (ActionSessionLog m, ActionPersist m) => AUID User -> m ()
+deleteUser uid = do
+    logEvent INFO $ cshow uid <> " is deactivated"
```
Comment (fisx) [inline src/Action.hs]: that sounds like the message i would expect if the user has been deactivated befre, and this action can't do it any more and is warning about it.  `deactivating user` is less confusing.  i'll change this.

Comment (fisx) [inline src/Action.hs]: or, even better, why don't we just `show` the acid-state transaction type?

Comment (fisx) [inline src/Action.hs]: ah, i guess because we have the transaction log for that.  (-:

File: src/Types/Prelude.hs [hunk 2]
```diff
@@ -78,6 +78,9 @@ type CSI' s a = CSI s s a a
 csi :: CSI s t a b => Iso s t a b
 csi = iso cs cs
 
+cshow :: (Show a, ConvertibleStrings String c) => a -> c
+cshow = cs . show
```
Comment (fisx) [inline src/Types/Prelude.hs]: I think we eliminated this a while back.  Why do you like names so much?  (-:

Comment (fisx) [inline src/Types/Prelude.hs]: anyway, we can keep it for a while.  (-:

File: src/Action.hs [hunk 3]
```diff
@@ -122,6 +123,7 @@ module Action
     , module Action.Smtp
     , sendMailToRole
 
+    , module Logger
```
Comment (fisx) [inline src/Action.hs]: just for the record: i also dislike re-exports.  they make it hard to get oriented, and i don't see a great benefit in using them.

File: src/Frontend/Core.hs [hunk 4]
```diff
@@ -676,7 +684,9 @@ coreRunHandler mp mr = do
                 AccessDenied s u -> handleDenied s u
                 AccessDeferred -> throwError500 "AccessDeferred should not be used with NotLoggedIn"
   where
-    handleDenied s u = throwServantErr $ (maybe Servant.err403 err303With u) { errBody = cs s }
+    handleDenied s u = do
+        logEvent WARN $ cs s
+        throwServantErr $ (maybe Servant.err403 err303With u) { errBody = cs s }
         -- FIXME log these events as INFO, should we do this here or more globally for servant errors.
```
Comment (fisx) [inline src/Frontend/Core.hs]: You should generally look out more for FXMEs near your code and update/remove them if appropriate.

**Comment by fisx** [comment]
Lots of noise from me (sorry), but LGTM.  This is a big improvement over the current status, thanks!

@andorp can be merged if you and travis are green.

**Comment by andorp** [comment]
LGTM

**Pull request merged**